crates/sui-indexer/src/tempdb.rs

```diff
@@ -0,0 +1,340 @@
+        if let Some(p) = &mut self.process {
+            match p.inner.try_wait() {
+                // This would mean the child process has crashed
+                Ok(Some(_)) => Err(HealthCheckError::NotRunning),
+
+                // This is the case where the process is still running
+                Ok(None) => pg_isready(self.port),
+
+                // Some other unknown error
+                Err(e) => Err(HealthCheckError::Unknown(e.to_string())),
+            }
+        } else {
+            Err(HealthCheckError::NotRunning)
+        }
+    }
+
+    fn wait_till_ready(&mut self) -> Result<(), HealthCheckError> {
+        let start = Instant::now();
+
+        while start.elapsed() < Duration::from_secs(10) {
+            match self.health_check() {
+                Ok(()) => return Ok(()),
+                Err(HealthCheckError::NotReady) => {}
+                Err(HealthCheckError::NotRunning | HealthCheckError::Unknown(_)) => break,
+            }
+
+            std::thread::sleep(Duration::from_millis(50));
+        }
+
+        Err(HealthCheckError::Unknown(
+            "timeout reached when waiting for service to be ready".to_owned(),
+        ))
+    }
+}
+
+#[derive(Debug)]
+enum HealthCheckError {
+    NotRunning,
+    NotReady,
+    #[allow(unused)]
+    Unknown(String),
+}
+
+/// Run the postgres `pg_isready` command to get the status of database
+///
+/// See <https://www.postgresql.org/docs/16/app-pg-isready.html> for more info
+fn pg_isready(port: u16) -> Result<(), HealthCheckError> {
+    let output = Command::new("pg_isready")
+        .arg("--host=localhost")
+        .arg("-p")
+        .arg(port.to_string())
+        .arg("--username=postgres")
+        .output()
+        .map_err(|e| HealthCheckError::Unknown(format!("command not found: pg_ctl: {e}")))?;
+
+    trace!("pg_isready code: {:?}", output.status.code());
+    trace!("pg_isready output: {}", output.stderr.escape_ascii());
+    trace!("pg_isready output: {}", output.stdout.escape_ascii());
+    if output.status.success() {
+        Ok(())
+    } else {
+        Err(HealthCheckError::NotReady)
+    }
+}
+
+/// Run the postgres `initdb` command to initialize a database at the provided path
+///
+/// See <https://www.postgresql.org/docs/16/app-initdb.html> for more info
+fn initdb(dir: &Path) -> Result<()> {
+    let output = Command::new("initdb")
+        .arg("-D")
+        .arg(dir)
+        .arg("--no-instructions")
+        .arg("--username=postgres")
+        .output()
+        .context("command not found: initdb")?;
+
+    if output.status.success() {
+        Ok(())
+    } else {
+        Err(anyhow!("unable to initialize database"))
+    }
+}
+
+/// Return an ephemeral, available port. On unix systems, the port returned will be in the
+/// TIME_WAIT state ensuring that the OS won't hand out this port for some grace period.
+/// Callers should be able to bind to this port given they use SO_REUSEADDR.
+pub fn get_available_port() -> u16 {
```

Comment:
throughout our tests we have a few places that do `("127.0.0.1:{}", get_available_port)`, what are your thoughts on having a function here that's like, `build_db_address(host: Option<u64>)` so we can reduce the manual `127.0.0.1`